src/webgpu/shader/execution/expression/call/builtin/textureNumLevels.spec.ts

```diff
@@ -34,12 +63,60 @@ Parameters
         'texture_2d_array',
         'texture_3d',
         'texture_cube',
-        'texture_cube_array`',
+        'texture_cube_array',
       ] as const)
       .beginSubcases()
-      .combine('sampled_type', ['f32-only', 'i32', 'u32'] as const)
+      .combine('sampled_type', ['f32', 'i32', 'u32'] as const)
+      .combine('view_type', ['full', 'partial'] as const)
+      // 1d textures can't have mipLevelCount > 0
+      .filter(t => t.texture_type !== 'texture_1d' || t.view_type !== 'partial')
   )
-  .unimplemented();
+  .beforeAllSubcases(t => {
+    t.skipIfTextureViewDimensionNotSupported(kTextureTypeToViewDimension[t.params.texture_type]);
+  })
+  .fn(t => {
+    const { texture_type, sampled_type, view_type } = t.params;
+    const { format } = kSampleTypeInfo[sampled_type];
+
+    const viewDimension = kTextureTypeToViewDimension[texture_type];
+    const dimension = getTextureDimensionFromView(viewDimension);
+    const isCube = texture_type.includes('cube');
+    const width = 64;
+    const height = dimension === '1d' ? 1 : width;
+    const depthOrArrayLayers = isCube ? 6 : 1;
```

Comment:
Isn't a depth of 1 problematic for mip count > 1 on 3D textures?

Reply:
a 3d texture can have depthOrArrayLayers = 1 and mipLevelCount = 1 + floor(log2(max(width, height, depthOrArrayLayers))). width is 64 so it has mip levels